saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
     std::cout << "}" << std::endl;
 }
 
-int main(int argc, char ** argv)
+/*
+ preprocess the input json file to make sure it's a valid json file.
+*/
+static int preProcessFile(std::string file_name)
+{
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        std::cerr << "Failed to open the input file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the input file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
```

Comment:
why limit like that ? i propose to make this a knob in config options via command line with default value 100M, but that could be configurable if needed, so no need for code change and recalculation will be needed if database will grow in size, btw. what is the size of db you tested on?

Reply:
In my development environment, it's 1.3M Bytes. I'm not sure what exactly size limit should be suitable for it. A configure file should be the best solution, but which config file should be used?

Reply:
Currently, the saidump doesn't read any config file in. If configurable is necessary, we need to create a config file or use an existed config file.

Reply:
no, no config file is necessary, i proposed to add a parameter to saidump executable to set maximum size from command line, not sure where you get this config idea?

Reply:
Good idea. I will do it.